pandas/core/exchange/column.py

```diff
@@ -0,0 +1,368 @@
+                      containing the data and whose second element is the data
+                      buffer's associated dtype.
+            - "validity": a two-element tuple whose first element is a buffer
+                          containing mask values indicating missing data and
+                          whose second element is the mask value buffer's
+                          associated dtype. None if the null representation is
+                          not a bit or byte mask.
+            - "offsets": a two-element tuple whose first element is a buffer
+                         containing the offset values for variable-size binary
+                         data (e.g., variable-length strings) and whose second
+                         element is the offsets buffer's associated dtype. None
+                         if the data buffer does not have an associated offsets
+                         buffer.
+        """
+        buffers: ColumnBuffers = {
+            "data": self._get_data_buffer(),
+            "validity": None,
+            "offsets": None,
+        }
+
+        try:
+            buffers["validity"] = self._get_validity_buffer()
+        except NoBufferPresent:
+            pass
+
+        try:
+            buffers["offsets"] = self._get_offsets_buffer()
+        except NoBufferPresent:
+            pass
+
+        return buffers
+
+    def _get_data_buffer(
+        self,
+    ) -> Tuple[PandasBuffer, Any]:  # Any is for self.dtype tuple
+        """
+        Return the buffer containing the data and the buffer's associated dtype.
+        """
+        if self.dtype[0] in (
+            DtypeKind.INT,
+            DtypeKind.UINT,
+            DtypeKind.FLOAT,
+            DtypeKind.BOOL,
+        ):
+            buffer = PandasBuffer(self._col.to_numpy(), allow_copy=self._allow_copy)
+            dtype = self.dtype
+        elif self.dtype[0] == DtypeKind.CATEGORICAL:
+            codes = self._col.values.codes
+            buffer = PandasBuffer(codes, allow_copy=self._allow_copy)
+            dtype = self._dtype_from_pandasdtype(codes.dtype)
+        elif self.dtype[0] == DtypeKind.STRING:
+            # Marshal the strings from a NumPy object array into a byte array
+            buf = self._col.to_numpy()
+            b = bytearray()
+
+            # TODO: this for-loop is slow; can be implemented in Cython/C/C++ later
+            for obj in buf:
+                if isinstance(obj, str):
+                    b.extend(obj.encode(encoding="utf-8"))
+
+            # Convert the byte array to a Pandas "buffer" using
+            # a NumPy array as the backing store
+            buffer = PandasBuffer(np.frombuffer(b, dtype="uint8"))
+
+            # Define the dtype for the returned buffer
+            dtype = (
+                DtypeKind.STRING,
+                8,
+                ArrowCTypes.STRING,
+                Endianness.NATIVE,
+            )  # note: currently only support native endianness
+        else:
```

Comment:
What about datetime dtype?

Reply:
It seems that it should be handled just like `int` and other primitive numpy types, so added there.

Reply:
distinct `self.dtype[0] == DtypeKind.DATETIME` branch is still present, Am I understand correctly that is should be removed since we handle datetime above in the common int-float-bool-datetime branch?

Reply:
@vnlitvinov, ping

Reply:
indeed, I've removed that branch in d1c0d56db0d4bd25fe0f88446e2498d7f270afe2